internal/dataapi/server/json.go

```diff
@@ -15,32 +15,25 @@
 package server
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"io"
 
 	"github.com/FerretDB/wire/wirebson"
-	"go.mongodb.org/mongo-driver/bson"
-	"go.mongodb.org/mongo-driver/bson/bsonrw"
+	"go.mongodb.org/mongo-driver/v2/bson"
 
 	"github.com/FerretDB/FerretDB/v2/internal/util/lazyerrors"
 )
 
 // marshalJSON encodes wirebson.RawDocument into extended JSON.
 func marshalJSON(raw wirebson.RawDocument, jsonDst io.Writer) error {
-	vw, err := bsonrw.NewExtJSONValueWriter(jsonDst, false, false)
+	b, err := bson.MarshalExtJSON(bson.Raw(raw), false, false)
 	if err != nil {
 		return lazyerrors.Error(err)
 	}
 
-	encoder, err := bson.NewEncoder(vw)
-	if err != nil {
-		return lazyerrors.Error(err)
-	}
-
-	encoder.IntMinSize()
-
-	err = encoder.Encode(bson.Raw(raw))
+	_, err = io.Copy(jsonDst, bytes.NewReader(b))
```

Comment:
Let's keep the implementation more closely to the example.
https://pkg.go.dev/go.mongodb.org/mongo-driver/v2/bson#example-Encoder-ExtendedJSON

I think we should be able to do something like below.
```
encoder := bson.NewEncoder(bson.NewExtJSONValueWriter(jsonDst, false, false))

if err := encoder.Encode(bson.Raw(raw)); err != nil {
	return lazyerrors.Error(err)
}
```
If encoder needs special option, it's simple to make the change :hugs:

Reply:
@KrishnaSindhur PTAL